lib/AST/TypeCheckRequests.cpp

```diff
@@ -157,6 +157,29 @@ void SuperclassTypeRequest::cacheResult(Type value) const {
     protocolDecl->LazySemanticInfo.SuperclassType.setPointerAndInt(value, true);
 }
 
+evaluator::DependencySource
+SuperclassTypeRequest::readDependencySource(Evaluator &e) const {
+  const auto access = std::get<0>(getStorage())->getFormalAccess();
+  return {
+    e.getActiveDependencySource(),
+    evaluator::getScopeForAccessLevel(access)
+  };
+}
+
+void SuperclassTypeRequest::writeDependencySink(Evaluator &eval,
+                                                ReferencedNameTracker &tracker,
+                                                Type value) const {
+  if (!value)
+    return;
+
+  // FIXME: This is compatible with the existing name tracking scheme, but
+  // ignoring this name when we fail to look up a class is bogus.
```

Comment:
This request is already suspicious since it uses the access level of the superclass to determine the kind of edge to write into the name trackers. It's doubly suspicious that it goes out of its way to *not* write an edge, regardless of why. 

When I wrote this, I considered these edges from extra typechecking sinks in isolation. In isolation, failing to register a potential member constraint, even on a type that doesn't exist, is a missed dependency. That's maybe being too conservative, but I'll take that over the alternative.